framework/app.mk

```diff
@@ -382,9 +382,14 @@ endif
 # Codesign command (OS X Only)
 codesign :=
 ifneq (,$(findstring darwin,$(libmesh_HOST)))
-	get_task_allow_entitlement := $(FRAMEWORK_DIR)/build_support/get_task_allow.plist
-	codesign := codesign -s - --entitlements $(get_task_allow_entitlement) $(app_EXEC)
+  ifneq (x$(MOOSE_NO_CODESIGN), x)
+    codesign :=
```

Comment:
You don't need this case at all. Note that we initialize it to empty before this condition. Just reverse your logic and drop this case.